phpBB/includes/ucp/ucp_pm_viewfolder.php

```diff
@@ -146,6 +137,13 @@ function view_folder($id, $mode, $folder_id, $folder)
 						break;
 					}
 				}
+				// Get avatars
```

Comment:
You should use the main query to fetch user data. This approach is wrong, it generates one query per every message. Fetch user data in the query starting from the line 536.

Reply:
This is supposed to use one query per message, as it gets the avatars for the left side, for every message title -> gets avatar -> templates

Reply:
What he is saying is instead use a koi on the original query to get the data rather than create a completely new one

Reply:
@hanakin @senky Okay can I do this or is there a simpler way
Get all the avatar info, message_id in single query, store it in an 2D- array
Then use the user_id to find the right avatar while templating.

Reply:
Oh I get it now. You can't use the main query. But you should use one query for all the users and then process them in php just like you suggest.

Reply:
Great!